src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -589,6 +589,13 @@ def miserable_users(org_id: int, metric_ids: Sequence[int], alias: str | None =
     )
 
 
+def non_negative_crash_free_rate(crash_rate, alias: str | None = None) -> Function:
+    # prevents crash_free_rate value to be negative when crash_rate > 1
+    # (this can happen due to a bug in other parts of the system)
+    # for more info see: https://github.com/getsentry/sentry/issues/73172
+    return Function("max2", [0, Function("minus", [1, crash_rate])], alias)
```

Comment:
`max2` is a Clickhouse function to return maximum of 2 elements: https://clickhouse.com/docs/en/sql-reference/functions/arithmetic-functions#max2